src/mailstoreobserver.cpp

```diff
@@ -357,14 +362,20 @@ void MailStoreObserver::updateMessages(const QMailMessageIdList &ids)
     // from read to unread ???
 
     foreach (const QMailMessageId &id, ids) {
+        const QMailMessageMetaData message(id);
         if (_publishedMessages.contains(id)) {
-            const QMailMessageMetaData message(id);
             // Check if message was read
             if (!notifyMessage(message)) {
                 _publishedMessages.remove(id);
                 _newMessages.remove(id);
-                _publicationChanges = true;
+            } else {
+                _updatedMessages.insert(id);
```

Comment:
We may not need this or the following branch, but I would prefer to leave them in, because a) they will be hit infrequently, and b) someone might add something to the updateNotifications() code that would require them to be added back.